Objects/listobject.c

```diff
@@ -1683,7 +1683,7 @@ binarysort(MergeState *ms, const sortslice *ss, Py_ssize_t n, Py_ssize_t ok)
     Py_ssize_t L, M, R;
 
     assert(0 <= ok && ok <= n && 1 <= n && n <= MAX_MINRUN);
-    /* assert s[:ok) is sorted */
+    /* assert s[:ok] is sorted */
```

Comment:
Is `s` a typo? I'd use `a` throughout. That is, `a[:ok]` here and also in the later invariants replace `0:L` with `a[0:L]` etc.

Reply:
Thanks. It's a typo.